README.md

```diff
@@ -2381,6 +2383,46 @@ Notice that the [`Browser`](#browser) is an immutable object, i.e. this
 method actually returns a *new* [`Browser`](#browser) instance with the
 given setting applied.
 
+#### withHeader()
+
+The `withHeader(string $header, string $value): Browser` method can be used to
+add a request header for all following requests.
+
+```php
+$browser = $browser->withHeader('User-Agent', 'ACME');
+
+$browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
+    //will contain the header User-Agent with the value ACME
+    var_dump($response->getHeaders());
+}, function (Exception $e) {
+    echo 'Error: ' . $e->getMessage() . PHP_EOL;
+});
+```
+
+Note that the new header will overwrite any headers previously set with
+the same name (case-insensitive). Following requests will use these headers
+by default unless they are explicitly set for any requests.
+
+#### withoutHeader()
+
+The `withoutHeader(string $header): Browser` method can be used to
+remove any default request headers previously set via
+the [`withHeader()` method](#withheader).
+
+```php
+$browser = $browser->withoutHeader('User-Agent');
+
+$browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
+    //won't contain the User-Agent header
+    var_dump($response->getHeaders());
```

Comment:
I think it can be confusing that we show two types of headers in this example (request and response).
```suggestion
    var_dump($response);
```